### ISSUE TITLE
No output from get_tuya_rtsp_stream.sh

### ISSUE THREAD

**Issue by linkian19** [felipecrs/get-tuya-webrtc-configs#2]
Link to the description of the issue can be found here: https://github.com/home-assistant/core/issues/68503#issuecomment-1973994779

Short synopsis: Filled in required values as documented, ensured that I had the IoT Live Video Stream API on my IoT Tuya account, ran the script, received error from openssl, installed openssl, ran script again, no output.

Will try to run script with extra flags as suggested and will update when I have the chance.

Thanks for all your work on this!

**Comment by felipecrs** [comment]
It's working just fine for me. And you don't need to modify anything in the script.

![image](https://github.com/felipecrs/get-tuya-webrtc-configs/assets/29582865/b2744b08-0c24-41ea-9512-547fc26d3ebc)

Perhaps some of your parameters are wrong? Also try with a camera you know that works first.

**Comment by linkian19** [comment]
I'll give this a try after the weekend when I have time to focus on the
computer. Will update then!

**Comment by felipecrs** [comment]
I pushed a change  https://github.com/felipecrs/get-tuya-webrtc-configs/commit/931fde84e1e959d51b129aa26387926649499683 to the script by the way, please try again.

**Comment by linkian19** [comment]
Awesome! I'll give that a try when I have the chance.

I was able to do a little digging this morning and was able to get a url and stream from one of my cameras (not the feit one yet) if I ran the script on a wsl instance I have. Still not receiving output from the script on my home assistant instance though.

From what I could tell the script is not generating the auth token you're looking for and I did notice that the 't' (assuming that's time) was getting a different result on my ha box vs my WSL instance.

If I took the final curl command from my WSL instance and ran it on ha I got a URL back. So I'm assuming it's something with my ha instance not working properly.

I'll dig into this more when I have a chance and will let you know!

Thanks again!

**Comment by felipecrs** [comment]
Got it. And by the way this script is just a proof of concept, I didn't polish it or anything for production use.

It works though in my HA instance within the VS Code add-on.

**Comment by linkian19** [comment]
So I think I've narrowed it down to the piece where it's getting the date/time.

For example: If I run this on my HA instance in the terminal I just get back: 1709

When I run it from other terminals I get: 1709<more numbers>

I verified this by copying the sent curl command to get the access_token and I did get an error back saying that the request time was invalid. Which seems pretty normal considering it's expecting some value and all I'm sending is 1709.

Now why my particular instance of HA is only giving me that? I haven't sorted out yet.

Of note: I am running HA on a VirtualBox VM, but I followed instructions according to HA guides and I haven't messed with anything else.

**Comment by linkian19** [comment]
Oh interesting! If I run it through the VS Code addon I will get a result. If I run it through the terminal, I don't. Hmmm.

**Comment by felipecrs** [comment]
If you were able to get the stream URL, now try playing with VLC.

If it plays, then you can probably get it to play in HA as well by using alexxit's webrtc integration with go2rtc.

**Comment by linkian19** [comment]
Hi there! Sorry I didn't update this sooner - I was able to get some output from this and I could test one of my cameras and if I opened it in VLC it would work, however, trying to implement it in go2rtc with the "echo" field did NOT seem to work. Not sure what the issue was there as I couldn't sort it out. 

I think this issue can probably be closed as the root of it was a difference between the ssh terminal and the VS Code addons (specifically with the "date" command).
